ons_ras_common/ons_rest_case.py

```diff
@@ -0,0 +1,244 @@
+#   Generic Configuration tool for Micro-Service environment discovery       #
+#   License: MIT                                                             #
+#   Copyright (c) 2017 Crown Copyright (Office for National Statistics)      #
+#                                                                            #
+##############################################################################
+#
+#   ONSCase wraps routines used to access the case service
+#
+##############################################################################
+
+
+class ONSCase(object):
+    """
+    This class is designed to take all the work out of accessing the case service. Initially it
+    should be able to validate and log events against the case service and also query the event
+    service for specific combinations of events. (for example to determine case status)
+    """
+    def __init__(self, env):
+        self._env = env
+        self._categories = None
+
+    def activate(self):
+        """"""
+        pass
+
+    def post_event(self, case_id, description=None, category=None, party_id=None, created_by=None, payload=None):
+        """
+        Post an event to the case service ...
+
+        :param case_id: The Id if the case to post against
+        :param description: Event description
+        :param category: Event category (must be a valid category)
+        :param party_id: Party Id
+        :param created_by: Who created the event
+        :param payload: An optional event payload
+        :return: status, message
+        """
+        #
+        #   Start by making sure we were given a working data set
+        #
+        if not (description and category and party_id and created_by):
+            msg = 'description={} category={} party_id={} created_by={}'.format(
+                description, category, party_id, created_by
+            )
+            self._env.logger.error('Insufficient arguments', arguments=msg)
+            return 500, {'code': 500, 'text': 'insufficient arguments'}
+
+        #
+        #   If this is our first time, we need to acquire the current set of valid categories
+        #   form the case service in order to validate the type of the message we're going to post
+        #
+        if not self._categories:
+            self._env.logger.info('first pass :: caching event category list')
+            categories = self._env.asyncio.access_endpoint('/categories')
+            if not categories:
+                return 404, {'code': 404, 'text': 'error loading categories'}
+            self._categories = {}
+            for cat in categories:
+                action = cat.get('name')
+                if action:
+                    self._categories[action] = cat
+                else:
+                    self._env.logger.warn('received unknown category "{}"'.format(str(cat)))
+            self._env.logger.info('first pass :: cached ({}) categories'.format(len(categories)))
```

Comment:
Why are we caching results on external endpoint outside of the http framework? Does the response have any cache-control headers? Why do we need to cache the response?